e2e/src/usecases/DappListRecent.js

```diff
@@ -0,0 +1,180 @@
+import { reloadReactNative } from '../utils/retries'
+import { sleep, waitForElementId, getElementText, getElementTextList } from '../utils/utils'
+
+const jestExpect = require('expect')
+
+/**
+ * From the home screen, navigate to the dapp explorer screen
+ * Disabled detox device sync due to dapp list double fetch / reanimated issue
+ * https://github.com/wix/Detox/issues/2799
+ */
+async function navigateToDappList() {
+  await device.disableSynchronization()
+  await waitForElementId('Hamburger')
+  await element(by.id('Hamburger')).tap()
+  await waitForElementId('dapps-explorer-icon')
+  await element(by.id('dapps-explorer-icon')).tap()
+  await waitFor(element(by.id('DAppExplorerScreen/loading')))
+    .not.toBeVisible()
+    .withTimeout(30 * 1000)
+  await sleep(5 * 1000)
+  await device.enableSynchronization()
+}
+
+/**
+ * From the drawer navigate to home screen
+ */
+async function navigateToHome() {
+  await waitForElementId('Hamburger')
+  await element(by.id('Hamburger')).tap()
+  await waitForElementId('Home')
+  await element(by.id('Home')).tap()
+}
+
+/**
+ * Scroll to a dapp in the dapp - iOS only
+ * @param {number} dappIndex: index of dapp to scroll to
+ */
+async function scrollToDapp(dappIndex = 0) {
+  try {
+    await waitFor(element(by.id('DappCard')).atIndex(dappIndex))
+      .toBeVisible(100)
+      .whileElement(by.id('DAppExplorerScreen/DappsList'))
+      .scroll(250, 'down')
+  } catch {
+    console.log('Catch of scrollToDapp')
+  }
+}
+
+export default DappListRecent = () => {
+  describe(':ios:', () => {
+    it('should show most recently used dapp leftmost on home screen', async () => {
+      // Get recently used dapps at start of test
+      const startRecentlyUsedDapps = await getElementTextList('RecentDapp-name')
+      const startRecentDappCount = startRecentlyUsedDapps.length
+
+      // If no recently used dapps check that RecentDapp element does not exist
+      if (startRecentDappCount === 0) await expect(element(by.id('RecentDapp'))).not.toExist()
+
+      // Navigate to DappList with sleep to allow for dapp list to load
+      await navigateToDappList()
+
+      // Scroll to first dapp or next after most recent dapp
+      await scrollToDapp(startRecentDappCount + 1)
+      await element(by.id('DappCard')).atIndex(startRecentDappCount).tap()
+
+      // Get dapp name in confirmation dialog
+      const dappPressed = await element(by.id('ConfirmDappButton')).getAttributes()
+      const dappPressedName = dappPressed.label.split('Go to ')[1]
+
+      // Confirm dapp open and navigate back to home
+      await element(by.id('ConfirmDappButton')).tap()
+      await element(by.text('Close')).tap()
+      await navigateToHome()
+
+      // Check that recently used dapp is now first in list
+      const endRecentlyUsedDapps = await getElementTextList('RecentDapp-name')
+      jestExpect(dappPressedName).toEqual(endRecentlyUsedDapps[0])
```

Comment:
maybe we can also add a `jestExpect(endRecentlyUsedDapps.length).toEqual(startRecentDappCount + 1)`?

Reply:
Added!